src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -112,6 +114,8 @@ export class DataView implements IIndexPattern {
     this.fieldAttrs = spec.fieldAttrs || {};
     this.allowNoIndex = spec.allowNoIndex || false;
     this.runtimeFieldMap = spec.runtimeFieldMap || {};
+    this.readableTitle = spec.readableTitle || '';
```

Comment:
Would it make more sense to call it `name` instead of `readableTitle`? I think `readableDescription` could also just be `description`. I don't think the existing `title` should stay as such since its so misleading ALTHOUGH I would keep it the same in the saved object. We could introduce an `indexPattern` attribute and then migrate code away from using `title`. Anyway, take these as suggestions rather than requirements for merging the PR.

Reply:
How about `label`? or `customLabel` like in https://github.com/elastic/kibana/pull/83717

Reply:
What is used for other saved objects? It seems there should be something common.

Reply:
++ on renaming `title` to `indexPattern`

Reply:
Added as a task in Jira to do a rename from title to indexPattern